src/runtime_src/core/pcie/tools/cloud-daemon/mpd_plugin.h

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (C) 2019 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+/* Declaring interfaces for mpd plugin. */
+/* For all functions return int, 0 = success, negative value indicate error. */
+
+#ifndef MPD_PLUGIN_H
+#define MPD_PLUGIN_H
+
+#include "xclbin.h"
+
+typedef int (*get_remote_msd_fd_fn)(size_t index, int &fd);
```

Comment:
Are all these `typedef` required?
I have the feeling you are re-inventing runtime polymorphism here.

Reply:
it is a habit how I write C code...